src/EditorFeatures/Core/Formatting/FormatCommandHandler.Paste.cs

```diff
@@ -54,9 +54,11 @@ private void ExecuteCommandWorker(PasteCommandArgs args, SnapshotPoint? caretPos
                 return;
 
             var subjectBuffer = args.SubjectBuffer;
-            var workspace = subjectBuffer.GetWorkspace();
-            if (workspace == null)
+            if (!subjectBuffer.TryGetWorkspace(out var workspace) ||
+                !workspace.CanApplyChange(ApplyChangesKind.ChangeDocument))
+            {
                 return;
+            }
```

Comment:
features (like command handlers) taht are already operating on mutable buffers are totally free to get back to their workspace to query things.  They are the point of integration with teh workspace and it does make sense for them to both query it to know what is allowed, and to directly ask it to make changes.

Reply:
makes sense 👍